### ISSUE TITLE
Early stopping based on evaluation loss

### ISSUE THREAD

**Issue by guillaumekln** [OpenNMT/OpenNMT-tf#128]
Early stopping would be very useful to remove the need to set a number of training steps. This should most likely be implemented as a hook.

There are currently talks to include this in TensorFlow: https://github.com/tensorflow/tensorflow/issues/18394

**Comment by thanish** [comment]
Hi, has the early stopping been brought into Onmt?

**Comment by guillaumekln** [comment]
Hi, no it is not (otherwise this issue would be closed).

It has been added to the Estimator API though. I'm still waiting for a PR of mine https://github.com/tensorflow/estimator/pull/22 that support stopping on any metrics.

**Comment by JordiCarreraVentura** [comment]
Hey! I'm really interested in this feature as well. I just noticed that other PR (tensorflow/estimator#22) that you @guillaumekln were waiting for seems to have already been merged

**Comment by guillaumekln** [comment]
Thanks for your interest.

With the transition to TensorFlow 2.0, we are planning to drop the Estimator integration in favor of custom training code for better control. In this context, I will most likely not work on this until this new code is released.

A PR to integrate this in the current code is still welcome.

**Comment by JordiCarreraVentura** [comment]
Ok, thanks a lot for the update!
